lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -412,6 +412,15 @@ def check_Parameter(o)
     if o.name =~ /^(?:0x)?[0-9]+$/
       acceptor.accept(Issues::ILLEGAL_NUMERIC_PARAMETER, o, :name => o.name)
     end
+    assignments =
+    if o.value.is_a?(Puppet::Pops::Model::AssignmentExpression)
+      [o.value]
+    else
+      o.value.eAllContents.select {|model| model.is_a? Puppet::Pops::Model::AssignmentExpression }
```

Comment:
o.value can apparently be nil at this point (this is why the tests are failing).